Svc/Framer/docs/sdd.md

```diff
@@ -1,46 +1,317 @@
+
+For an example of setting up a `Framer` instance, see the
+`downlink` instance in [`Ref/Top/instances.fpp`](../../../Ref/Top/instances.fpp).
+
+### 4.7. Port Handlers
+
+#### 4.7.1. comIn
+
+The `comIn` port handler receive a reference to an `Fw::Com` buffer _B_
+and an integer context value.
+It calls the `frame` method of `m_protocol`, passing in the
+address and length of _B_ and the packet type
+`Fw::ComPacket::FW_PACKET_UNKNOWN`.
+
+#### 4.7.2. bufferIn
+
+The `bufferIn` port handler receives a reference to an  `Fw::Buffer`
+object _B_.
+It calls the `frame` method of `m_protocol`, passing in the
+data address and size of _B_ and the packet type
+`Fw::ComPacket::FW_PACKET_FILE`.
+
+<a name="fpi-impl"></a>
+### 4.8. Implementation of Svc::FramingProtocolInterface
+
+<a name="allocate"></a>
+#### 4.8.1. allocate
+
+The implementation of `allocate` invokes `framedAllocate`.
+
+<a name="send"></a>
+#### 4.8.2. send
+
+The implementation of `send` takes a reference to an `Fw::Buffer`
+_B_ representing framed data and does the following:
+
+1. Invoke `framedOut`, passing in _B_ as the argument.
+
+1. Check the return status of the invocation. 
+If the return status is not `Drv::SendStatus::SEND_OK`, then
+use `Fw::Logger::logMsg` to log an error message.
+Don't send an event report in this case, because downlink is
+apparently not working.
+
+## 5. Ground Interface
+
+None.
+
+If an error occurs, `Framer` writes to the system log.
+The rationale is that if something is wrong with the framing, then
+downlink of events is unlikely to work.
+
+## 6. Example Uses
+
+<a name="top-diagrams"></a>
+### 6.1. Topology Diagrams
+
+The following topology diagrams show how to connect `Svc::Framer`
+to a telemetry database, an event collector, a file downlink component,
+and a byte stream driver.
+The diagrams use the following instances:
 
-![classdiagram](./img/class_diagram_framer.png)
+* `comm`: An instance of
+[`Drv::ByteStreamDriverModel`](../../../Drv/ByteStreamDriverModel/docs/sdd.md), for example
+[`Drv::TcpClient`](../../../Drv/TcpClient/docs/sdd.md).
 
-## Requirements
+* `buffMgr`: An instance of [`Svc::BufferManager`](../../BufferManager/docs/sdd.md)
 
-| Name | Description | Validation |
-|---|---|---|
-| TBD | TBD | TBD |
+* `fileDownlink`: An instance of [`Svc::FileDownlink`](../../FileDownlink/docs/sdd.md).
+
+* `framer`: An instance of `Svc::Framer`.
+
+* `chanTlm`: An instance of [`Svc::TlmChan`](../../TlmChan/docs/sdd.md).
+
+* `eventLogger`: An instance of [`Svc::ActiveLogger`](../../ActiveLogger/docs/sdd.md).
+
+**Topology 1: Telemetry packets:**
+
+<div>
+<img src="img/top/tlm.png" width=1000/>
+</div>
+
+The `chanTlm` instance sends telemetry packets to the `framer` instance.
+
+**Topology 2: Event packets:**
+
+<div>
+<img src="img/top/event.png" width=1000/>
+</div>
+
+The `eventLogger` instance sends event packets to the `framer` instance.
+
+**Topology 3: File packets:**
+
+<div>
+<img src="img/top/file.png" width=1000/>
```

Comment:
Fwiw, afaict this file (`file.png`) is ambiguous and broke the autodocumentation (which flattens files into a single directory).

I've posted #1599 which renames it. (I can't hang a comment on an image file, so I'm hanging it here where it's being used.)